cargo-dylint/tests/ci.rs

```diff
@@ -133,6 +133,165 @@ fn cargo_dylint_and_dylint_readmes_are_equal() {
     compare_lines(&cargo_dylint_readme, &dylint_readme);
 }
 
+#[test]
+fn examples_readme_contents() {
+    let examples_dir = find_examples_dir();
+    let categories = vec![
+        "general",
+        "supplementary",
+        "restriction",
+        "experimental",
+        "testing",
+    ];
+    
+    // Read the existing README
+    let readme_path = examples_dir.join("README.md");
+    let readme_content = read_to_string(&readme_path).unwrap();
+
+    // Generate just the lint description tables
+    let tables = generate_lint_tables(&examples_dir, &categories);
+    
+    // Extract the current tables section from README using markers
+    let current_tables = extract_between_markers(&readme_content, 
+        "<!-- lint descriptions start -->",
+        "<!-- lint descriptions end -->"
+    ).unwrap_or_else(|| panic!("Lint description markers not found in README.md"));
+
+    // Extract and compare just the meaningful content (links and descriptions)
+    let expected_content = extract_meaningful_content(&tables);
+    let actual_content = extract_meaningful_content(&current_tables);
+
+    assert_eq!(
+        expected_content,
+        actual_content,
+        "Lint descriptions in README.md do not match expected content"
+    );
+}
+
+fn extract_meaningful_content(s: &str) -> Vec<(String, String)> {
+    let link_re = Regex::new(r"\[`([^`]+)`\]\([^)]+\)").unwrap();
+    let desc_re = Regex::new(r"\|[^|]*\| ([^|]+) \|").unwrap();
+    
+    let mut result = Vec::new();
+    let mut lines = s.lines().peekable();
+    
+    while let Some(line) = lines.next() {
+        if let (Some(link_cap), Some(desc_cap)) = (link_re.captures(line), desc_re.captures(line)) {
+            result.push((
+                link_cap.get(1).unwrap().as_str().to_string(),
+                desc_cap.get(1).unwrap().as_str().trim().to_string()
+            ));
+        }
+    }
+    
+    result.sort();
+    result
+}
+
+fn generate_lint_tables(examples_dir: &Path, categories: &[&str]) -> String {
```

Comment:
As per the comment above regarding `assert_eq!`, this function will need to be adjusted to get the spacing right.

I recommend computing all of the formatted "Example" and "Description/check" entries, and then determining the columns' widths from that.

Something similar to this: https://github.com/trailofbits/dylint/blob/67eddafd13de22a77b3a90922f74f06742cb8174/dylint/src/lib.rs#L182-L193